### PR TITLE
Cr 959 address type not applicable

### PR DESCRIPTION
**Pull request by cookel2**
# Motivation and Context

This change is required in order to make sure that the getCaseByUprn endpoint will not return any case that has an addressType of "NA" but instead will return a '404 Not Found' error.
NB. Examples of valid addresses that have addressType=NA are addresses for things like electricity transformers or public telephone boxes e.g.

http://localhost:8171/cases/uprn/10025220117
would previously bring back a case like this:
{
        "id": "e689c3bd-579c-4e30-848c-04d4b9df7d07",
        "caseRef": null,
        "caseType": null,
        "addressType": "NA",
        "estabType": "OTHER",
        "estabDescription": "NA",
        "allowedDeliveryChannels": [
            "POST",
            "SMS"
        ],
        "createdDateTime": null,
        "addressLine1": "Public Telephone 13M From 11 Nine Acres On Unnamed Road",
        "addressLine2": "Steep Marsh Bungalows",
        "addressLine3": "Steep",
        "townName": "Petersfield",
        "region": "E",
        "postcode": "GU32 2BW",
        "uprn": 10025220117,
        "estabUprn": null,
        "caseEvents": null,
        "secureEstablishment": false
    }

but instead now it returns this:
{
    "error": {
        "code": "RESOURCE_NOT_FOUND",
        "timestamp": "20200521161454042",
        "message": "AddressType of 'NA' not valid for Census"
    }
}

However, I have discovered that the functionality only works if a case, with the relevant uprn, and addressType=NA, does not already exist in Firestore. Otherwise the getCaseByUprn endpoint will find the case in Firestore and return it (instead of the expected 404 error). If that happens, as it was in my local environment, then the new test will fail.

The following new scenario is tested in the testCaseEndpoints feature:

Scenario: [CR-T377] AddressType Not Applicable
  Given the CC agent has selected an address that is not of addressType CE, HH, or SPG
  And the case service does not have any case created for the address in question
  When Get/Case API returns a 404 error because there is no case found
  Then the CC SVC must also return a "404 Not Found" error

NB. The above scenario has been successfully tested in census-integration-dev:
https://concourse.census-gcp.onsdigital.uk/teams/int/pipelines/integration-services/jobs/cccuc-dev-unreleased-acceptance-tests/builds/83

# What has changed

The testCaseEndpoint feature has had scenario CR-T377 added to it.

# How to test?

Run the contact centre cucumber tests locally by following the instructions in this link:
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=SDC&title=How+to+Run+CCCUC+Locally

### CODE REVIEW

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/address/TestAddressEndpoints.java [hunk 1]
```diff
@@ -128,4 +143,94 @@ public void an_empty_list_of_addresses_for_my_search_is_returned() {
           "Address list size must be zero", 0, addressQueryResponseDTO.getAddresses().size());
     }
   }
+
+  @Given("the respondent calls the CC with a fulfilment request")
+  public void the_respondent_calls_the_CC_with_a_fulfilment_request() {
+    log.info("Nothing to do here: the respondent calls the CC with a fulfilment request");
```
Comment (andrewjohnysons) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/address/TestAddressEndpoints.java:149]: Don't see th epoint of having a step that doesn't do anything #justsaying

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/cases/TestCaseEndpoints.java [hunk 2]
```diff
@@ -800,6 +827,217 @@ public void cc_Advisor_selects_the(String statusSelected) {
...
+    assertNull(newAddress.getSourceCaseId());
+
+    collectionCase = newAddress.getCollectionCase();
+    assertNotNull(collectionCase.getId());
+    assertNull(collectionCase.getCaseType());
+    assertEquals("CENSUS", collectionCase.getSurvey());
+    assertNull(collectionCase.getFieldCoordinatorId());
+    assertNull(collectionCase.getFieldOfficerId());
+
+    address = collectionCase.getAddress();
```
Comment (andrewjohnysons) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/cases/TestCaseEndpoints.java:1025]: can this address be null?

File: src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/cases/TestCaseEndpoints.java [hunk 3]
```diff
@@ -859,4 +1097,95 @@ public void aSurveyLaunchedEventIsEmittedToRMWhichContainsTheLaunchStatusType()
...
+        getRestTemplate()
+            .exchange(
+                builder.build().encode().toUri(),
+                HttpMethod.GET,
+                null,
+                new ParameterizedTypeReference<AddressQueryResponseDTO>() {});
+
+    log.with(addressQueryResponse).info("The address query response here");
+
+    AddressQueryResponseDTO addressQueryBody = addressQueryResponse.getBody();
```
Comment (andrewjohnysons) [inline src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/cases/TestCaseEndpoints.java:1119]: null response?

**Comment by cookel2** [comment]
Closing this PR as it is superceded by this one:
https://github.com/ONSdigital/census-contact-centre-cucumber/pull/39

**Pull request closed without merging**